### PR TITLE
Enable support for full-swing videos

### PR DESCRIPTION
**Pull request by JanuszL**
- enables detection if the decoded video uses full 0-255
   value range or just 16-235 and convert to RGB accordingly

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
- enables detection if the decoded video uses full 0-255
   value range or just 16-235 and convert to RGB accordingly
- related to https://github.com/NVIDIA/DALI/issues/4441

### Affected modules and functionalities:
- video reader
- GPU frames decoder

### Key points relevant for the review:
- conversion coefficients

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
    - test_video.py
      - test_full_range_video
      - test_full_range_video_in_memory
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Review by szalpal — APPROVED** [review]

File: dali/operators/reader/nvdecoder/imgproc.cu [hunk 1]
```diff
@@ -41,6 +41,21 @@ __constant__ float ycbcr2rgb_mat[9] = {
   1.164383f * 255.0f,  2.017232f * 255.0f,  0.0f
 };
 
+
+// https://en.wikipedia.org/wiki/YUV#Y%E2%80%B2UV444_to_RGB888_conversion
+__constant__ float ycbcr2rgb_mat_norm_full_range[9] = {
```
Comment (mzient) [inline dali/operators/reader/nvdecoder/imgproc.cu:46]: I guess we could use color space conversion functions - though it may be a bit out of scope.

Comment (JanuszL) [reply dali/operators/reader/nvdecoder/imgproc.cu:46]: We have this strange float output path. I would move this rework to another PR.

**Review by mzient — APPROVED** [review]
Caveat: I'm a co-author, so I think another pair of eyes is warranted.

File: dali/operators/reader/nvdecoder/imgproc.cu (hunk 1 above)
Comment (JanuszL) [reply dali/operators/reader/nvdecoder/imgproc.cu:46]: Done

**Review by mzient — APPROVED** [review]

File: dali/operators/reader/nvdecoder/imgproc.cu [hunk 2]
```diff
@@ -107,22 +40,37 @@ __global__ void process_frame_kernel(
...
+  vec<3, float> yuv;
+  const float scale = 255.0f/256;
+  yuv[0] = tex2D<float>(luma, src_x, src_y);
   auto cbcr = tex2D<float2>(chroma, src_x * 0.5f, src_y * 0.5f);
-  ycbcr.cb = cbcr.x;
-  ycbcr.cr = cbcr.y;
+  yuv[1] = cbcr.x;
+  yuv[2] = cbcr.y;
 
+  yuv = yuv * scale;
```
Comment (mzient) [inline dali/operators/reader/nvdecoder/imgproc.cu:50]: Perhaps for readability it'd be better to define the scale right here.
```suggestion
  const float scale = 255.0f/256;
  yuv = yuv * scale;
```

Comment (JanuszL) [reply dali/operators/reader/nvdecoder/imgproc.cu:52]: Done

File: dali/operators/reader/nvdecoder/imgproc.cu (hunk 1 above)
Comment (JanuszL) [reply dali/operators/reader/nvdecoder/imgproc.cu:46]: Reverted due to subtle discrepancies.

**Pull request merged**